module/VuFind/src/VuFind/Content/PageLocator.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * Class PageLocator
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Moravian Library 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content;
+
+/**
+ * Class PageLocator
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class PageLocator
+{
+    /**
+     * Types/formats of content
+     *
+     * @var array $types
+     */
+    protected $types = [
+        'phtml',
+        'md',
+    ];
+
+    /**
+     * Theme info service
+     *
+     * @var \VuFindTheme\ThemeInfo
+     */
+    protected $themeInfo;
+
+    /**
+     * Translator
+     *
+     * @var \Laminas\Mvc\I18n\Translator
+     */
+    protected $translator;
+
+    /**
+     * Default language
+     *
+     * @var string
+     */
+    protected $defaultLanguage;
+
+    /**
+     * Page constructor.
+     *
+     * @param \VuFindTheme\ThemeInfo       $themeInfo  Theme information service
+     * @param \Laminas\Mvc\I18n\Translator $translator Translator
+     * @param \Laminas\Config\Config       $config     Main configuration
+     */
+    public function __construct($themeInfo, $translator, $config)
+    {
+        $this->themeInfo = $themeInfo;
+        $this->translator = $translator;
+        $this->defaultLanguage  = $config->Site->language;
+    }
+
+    /**
+     * Try to find template information about desired page
+     *
+     * @param string $pathPrefix Subdirectory where the template should be located
+     * @param string $pageName   Template name
+     *
+     * @return array|null Null if template is not found or array with keys renderer
+     *                    (type of template), path (full path of template), page
+     *                    (page name)
+     */
+    public function determineTemplateAndRenderer($pathPrefix, $pageName)
+    {
+        $language = $this->translator->getLocale();
```

Comment:
Since we are only using the translator to obtain the current locale, we probably don't need to put a translator dependency on this class; instead, it might make more sense to do the locale lookup in the factory and have the class require only the locale string. This will make unit test development simpler.

Reply:
Done